examples/stream/port_duplex.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const { Duplex } = require('stream');
+
+class PortDuplex extends Duplex {
+  #port = undefined;
+  #transfer = false;
+
+  constructor (port, options) {
+    const {
+      readable = true,
+      writable = true,
+      transfer = false
+    } = { ...options };
+    if (typeof readable !== 'boolean') {
+      throw new TypeError('readable must be a boolean');
+    }
+    if (typeof writable !== 'boolean') {
+      throw new TypeError('writable must be a boolean');
+    }
+    if (typeof transfer !== 'boolean') {
+      throw new TypeError('transfer must be a boolean');
+    }
+    super({ autoDestroy: true, readable, writable });
+    this.#port = port;
+    this.#transfer = transfer;
+    this.#port.onmessage = PortDuplex.#onmessage.bind(this);
+  }
+
+  _write (chunk, encoding, callback) {
+    if (typeof chunk === 'string') {
+      chunk = Buffer.from(chunk, encoding);
+    }
+    const transferList = this.#transfer ? [chunk.buffer] : undefined;
+    this.#port.postMessage(chunk, transferList);
+    callback();
+  }
+
+  _final (callback) {
+    this.#port.postMessage(null);
+    callback();
+  }
+
+  _read () {
+    // Do nothing here. A more complete example would
+    // implement proper read/pause behavior.
+  }
```

Comment:
Btw, I’ve thought about porting the code from our internal Worker stdio implementation to npm. This is definitely giving more motivation for that :)

Reply:
Likely would be good I think. I'm considering the possibility of Piscina providing a more complete version of `PortDuplex` out of the box.

Reply:
@jasnell The thing is, in the end streaming use cases for Workers are even rarer than use cases for Workers in general. I don’t think we should encourage this pattern if users don’t really need it.

Reply:
That's absolutely fair. The key reason I'm thinking through this is to explore the various good and bad ways of using the workers and what the various caveats are.